src/parser/jobs/pld/modules/Atonement.tsx

```diff
@@ -38,34 +36,52 @@ export default class Atonement extends Analyser {
 		}, () => this.stacksUsed++)
 
 		this.addEventHook({
-			type: 'combo',
-			source: this.parser.actor.id,
-			action: this.data.actions.ROYAL_AUTHORITY.id,
-		}, this.onGenerateStacks)
+			type: 'statusApply',
+			target: this.parser.actor.id,
+			status: this.data.statuses.SWORD_OATH.id,
+		}, this.onApplySwordOath)
+		this.addEventHook({
+			type: 'statusRemove',
+			target: this.parser.actor.id,
+			status: this.data.statuses.SWORD_OATH.id,
+		}, this.onRemoveSwordOath)
 
 		this.addEventHook('complete', this.onComplete)
 	}
 
-	private onGenerateStacks(): void {
-		// Tracking overcap in case someone wants to do something with it later
-		this.overcap += Math.max(this.currentStacks + STACKS_GAINED - MAX_STACKS, 0)
-		this.potentialStacks += STACKS_GAINED
-		this.currentStacks = Math.min(this.currentStacks + STACKS_GAINED, MAX_STACKS)
+	private onApplySwordOath(event: Events['statusApply']): void {
+		if (event.data == null) { return }
+
+		// Track potential uses & any overcap due to reapplication while the status was still active
+		if (event.data === this.data.statuses.SWORD_OATH.stacksApplied) {
+			this.overcap += this.currentStacks
+			this.potentialStacks += this.data.statuses.SWORD_OATH.stacksApplied
+		}
+
+		this.currentStacks = event.data
+	}
+
+	private onRemoveSwordOath(): void {
+		// If any stacks were left when the status fell off, keep track of those
+		if (this.currentStacks > 0) {
+			this.droppedStacks += this.currentStacks
+		}
```

Comment:
This counts ALL removals of the last stack as drops, whether they were expired or used.  We don't get an Apply stacks: 0 event on the last usage to update the currentStacks value.

This log http://127.0.0.1:3000/fflogs/gqaw9kRchv67W2xP/11/2 has a timeout at 2:19 for testing/comparison, but should only show 3 dropped stacks (2 timed out at 2:19 and 1 left over at end of fight)

Reply:
Oop, you're right. I don't feel like having leftover stacks at the end of the fight should really count as a drop, but I corrected the stack usage/statusRemoval tracking to make sure we're tracking only the unused stacks at the time the status falls off.